tests/test_hpack.c

```diff
@@ -701,3 +701,60 @@ static int test_hpack_dynamic_table_get(struct aws_allocator *allocator, void *c
     aws_http_library_clean_up();
     return AWS_OP_SUCCESS;
 }
+
+static int s_check_header(
```

Comment:
we we already have negative test cases? I see some failure scenarios above (other than for allocation failures) that need testing.

Reply:
I am not sure... What kind of failure scenarios besides this one?